kiosk/sources/extensions/kiosk_metadata.move

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// This module defines a `kiosk_metadata` extension 
+// that can be used to add metadata to a kiosk.
+// This metadata can be used to store information
+// about the kiosk, such as a name, an image_url or extra key, value pairs.
+module kiosk::kiosk_metadata {
+
+    use sui::object::{UID, ID};
+    use sui::kiosk::{Self, Kiosk, KioskOwnerCap};
+    use std::string::{String};
+    use sui::vec_map::{Self, VecMap};
+    use sui::dynamic_field as df;
+    use std::vector::{Self};
+    use sui::event::{emit as emit_event};
+
+    struct Metadata has store, copy, drop {
+        metadata: VecMap<String, String>
+    }
+
+    // === Events === 
+    // Event that is emmitted when kiosk metadata is created
+    struct KioskMetaDataCreated has copy, drop {
+        kiosk_id: ID
+    }
+
+    // the kiosk & kioskOwnerCap do not match, the user is not the owner of the kiosk.
+    const ENotKioskOwner: u64 = 1;
+    // invalid length of the fields and values vectors for key,val inserts.
+    const EVecLengthMismatch: u64 = 2;
+    // trying to remove a key that doesn't exist
+    const EKeyNotExists: u64 = 3;
+
+    struct Kiosk_MetadataExt has copy, store, drop {}
+
+    // Create a new Metadata object for the user's Kiosk.
+    // Creates a dynamic field object with the Metadata.
+    entry public fun enable(kiosk: &mut Kiosk, cap: &KioskOwnerCap){
```

Comment:
```suggestion
    public fun enable(kiosk: &mut Kiosk, cap: &KioskOwnerCap){
```

Entry functions should not be used (I know it's not obvious yet), they limit what a function can do in a PTB.